### DISCUSSION TITLE
Pipeline should fail on Renovate error

### DISCUSSION THREAD

**Discussion by schamal2205** [renovatebot/renovate#20770] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.157.1

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Pipeline Step should fail if Renovate has an Error. 

![grafik](https://user-images.githubusercontent.com/37846690/223067583-1e98cef5-4142-48a0-a5a2-bdf5c2f9ce22.png)

If the pipeline is not failing, i do not get notified that e.g. manual work has to be done, to get the package update working. 
Without the notification, important updates may be missed or overlooked.

### What you tried so far.

I set HostRules to Abort on Error = true: 
```
hostRules: [
        {
            hostType: 'nuget',
            matchHost: 'https://pkgs.dev.azure.com',
            username: '',
            password: process.env.TOKEN,
        },
        {
           abortOnError: true,
        }
    ],
```

And the environment Variable RENOVATE_X_HARD_EXIT to true. 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
External host errors are often transient and therefore do not log an _error_, only a _warning_. So the system is behaving as designed right now.

Default behavior cannot change because it would impact the hosted app too much, but you are welcome to think of ideas for something you could opt into, such as:
- Exit Renovate error for any warning, not just error, or
- Log external host errors as errors, not warnings, 
- etc.

**Reply by schamal2205** [reply]
Ok got it. 
I understand why the setting is like it is. 
So for me as a workaround, i'm parsing now the Renovate-output for Errors and let the pipeline fail if there are any. 
I think this would be "Log warning as Error"-setting.
